### PR TITLE
Encoders now automatically have Distance Per Pulse as a property.

### PR DESCRIPTION
**Pull request by JohnGilb**

### CODE REVIEW

File: src/xbot/common/controls/sensors/wpi_adapters/EncoderWPIAdapter.java [hunk 1]
```diff
@@ -2,30 +2,28 @@
...
 public class EncoderWPIAdapter implements XEncoder {
 
     Encoder adapter;
     boolean inverted;
+    public DoubleProperty distancePerPulse;
 
-    public EncoderWPIAdapter(int aChannel, int bChannel) {
+    public EncoderWPIAdapter(String name, int aChannel, int bChannel, double defaultDistancePerPulse, XPropertyManager propMan) {
         adapter = new Encoder(aChannel, bChannel);
+        distancePerPulse = propMan.createPersistentProperty(name + "-DistancePerPulse", defaultDistancePerPulse);
```
Comment (WasabiFan) [inline src/xbot/common/controls/sensors/wpi_adapters/EncoderWPIAdapter.java]: Spaces! Put spaces in property names!

Comment (JohnGilb) [inline src/xbot/common/controls/sensors/wpi_adapters/EncoderWPIAdapter.java]: rather than pretty dashes? =] @WasabiFan

Comment (JohnGilb) [inline src/xbot/common/controls/sensors/wpi_adapters/EncoderWPIAdapter.java]: camel casing here seems appropriate overall though - space on the dashboard is at a slight premium, especially with how many properties we have.

Comment (WasabiFan) [inline src/xbot/common/controls/sensors/wpi_adapters/EncoderWPIAdapter.java]: I see the point on space availability, but I think that readability is a very important factor to consider; humans read space-separated words much more easily than a single block of characters, even if each word is capitalized. Prioritizing the 1/8 space savings over the ability to glance at properties to read then seems like a pretty bad tradeoff.

**Pull request merged**